In [ ]:
%matplotlib inline
import numpy as np
import matplotlib as plt

## 第一題：

題目：給定一個矩陣m，裡面的值都大於0。
請將矩陣m裡大於等於1的數字都變成1，小於1的變成0。
m = np.array([[0,1,10],[2,6,7]])
提示：使用巢狀迴圈(nested loop)加判斷式

In [ ]:
m = np.array([[0,1,10],[2,6,7]])

In [4]:
m

array([[ 0,  1, 10],
       [ 2,  6,  7]])

In [17]:
for i in range(m.shape[0]):
    for j in range(m.shape[1]):
        if m[i,j] >= 1:
            m[i,j] = 1
        elif m[i,j] < 1:
            m[i,j] = 0

In [18]:
m

array([[0, 1, 1],
       [1, 1, 1]])

## 第二題：
有A,B,C三個班級，每班皆有10個學生，期中考結束後，老師希望計算各班級每位學生的加權平均成績。
請隨機產生3個班級的成績單,每班10人,共五科,權重為weights=[0.25,0.25,0.25,0.15,0.10]。
請算出班級加權平均成績，並以一個array輸出。

In [40]:
A = np.random.rand(50).reshape(10,5)*100
B = np.random.rand(50).reshape(10,5)*100
C = np.random.rand(50).reshape(10,5)*100
weights=[0.25,0.25,0.25,0.15,0.10]

In [57]:
SA = np.sum([A*weights], axis=2)
SB = np.sum([B*weights], axis=2)
SC = np.sum([C*weights], axis=2)
ALLscore = np.array([SA , SB , SC])

In [58]:
ALLscore

array([[[ 16.4349858 ,  40.60184235,  31.3306609 ,  64.66267722,
          45.00703852,  59.41127362,  27.07193189,  77.71572767,
          62.34419558,  43.3117474 ]],

       [[ 47.56751714,  49.69573419,  24.09845897,  49.68640057,
          75.08142009,  44.27854822,  58.35749028,  69.40301817,
          65.89946659,  36.12596499]],

       [[ 55.09133433,  43.69851341,  40.9130182 ,  55.51998938,
          66.64354144,  43.8273956 ,  18.1181296 ,  55.9252557 ,
          45.45301457,  35.01408726]]])

## 挑戰題一：

請定義一個全班高微成績的函式，輸入是一個(3, 30, 3)的array，  
表示3個班級中30位學生的成績(小考，期中考，期末考)  。

目標輸出則是一個(3, 30, 1)的array，表示所有學生的期末成績，
期末成績計算方式如下：

1. 小考20%，期中考35%，期末考45%  
2. 期中考40%，期末考60%。  
擇優作為期末成績